### PR TITLE
solution

### PR DESCRIPTION
**Pull request by lefteris-L**

### CODE REVIEW

File: src/makeInfinityAdder.js [hunk 1]
```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let tempSum = 0;
+  let sum;
+
+  return function add(n) {
+    if (isNaN(n)) {
```
Comment (ysrckr) [inline src/makeInfinityAdder.js:13]: I liked this approach. You can add `&& n === undefined` so `adder(2)('str')` doesn't return a value.

File: src/makeInfinityAdder.js [hunk 2]
```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let tempSum = 0;
+  let sum;
```
Comment (ysrckr) [inline src/makeInfinityAdder.js:10]: you can set `let sum = 0` and inside the function you can set `
let tempSum  =  sum
sum = 0
return tempSum
`
with this you can have only one global variable.

**Review by ysrckr — APPROVED** [review]
Just some suggestion but everything looks ok.

**Review by meforoma — APPROVED** [review]
working solution 👍

**Pull request closed without merging**